### PR TITLE
Move resize and persist

### PR DESCRIPTION
**Pull request by spmcbride1201**
Now able to move and resize charts and persist dashboard data to Redux and local storage.

### CODE REVIEW

File: src/components/GMGrid.js [hunk 1]
```diff
@@ -21,16 +24,43 @@ class GMGrid extends Component {
...
 
-  render() {
-    const { dashboards, metrics } = this.props;
+  state = {
+    name: ""
+  }
+
+  componentWillMount() {
+    this.setName(this.props);
+  }
```
Comment (tilleryd) [inline src/components/GMGrid.js]: `\n`

File: src/components/GMGrid.js [hunk 2]
```diff
@@ -21,16 +24,43 @@ class GMGrid extends Component {
...
+  componentWillReceiveProps(nextProps) {
+    this.setName(nextProps);
```
Comment (tilleryd) [inline src/components/GMGrid.js]: Perhaps check for `this.props.name !== nextProps.name` to prevent a re-render.

File: src/json/dashboards.json [hunk 3]
```diff
@@ -189,20 +182,26 @@
...
+          { "i": "Heap",                "x": 0, "y": 0, "w": 6, "h": 9, "minW": 4, "minH": 5 },
+          { "i": "Classes",             "x": 6, "y": 0, "w": 6, "h": 9, "minW": 4, "minH": 5 }
+        ],
+        "md": [
+          { "i": "Heap",                "x": 0, "y": 0, "w": 6, "h": 9, "minW": 4, "minH": 5 },
+          { "i": "Classes",             "x": 6, "y": 0, "w": 6, "h": 9, "minW": 4, "minH": 5 }
+        ],
+        "sm": [
+          { "i": "Heap",                "x": 0, "y": 0, "w": 6, "h": 9, "minW": 4, "minH": 5 },
+          { "i": "Classes",             "x": 6, "y": 0, "w": 6, "h": 9, "minW": 4, "minH": 5 }
```
Comment (tilleryd) [inline src/json/dashboards.json]: a lot of whitespace here

Comment (tilleryd) [inline src/json/dashboards.json]: oh i see, you're lining it up with other others, i guess it's not a big issue

File: src/store.js [hunk 4]
```diff
@@ -1,18 +1,25 @@
+import axios from 'axios';
```
Comment (tilleryd) [inline src/store.js]: What are the benefits of using `axios` over `whatg-fetch`?

File: src/store.js [hunk 5]
```diff
@@ -118,6 +165,17 @@ Hook((action, getState) => {
...
+// Persist dashboards to localStorage 
+Hook((action, getState) => {
+  if (action.type === 'setInterval') {
+    Actions.stopPolling();
+    Actions.startPolling({
+      endpoints: getState().settings.metricsEndpoints,
+      interval: getState().settings.interval
+    });
+  }
+});
```
Comment (tilleryd) [inline src/store.js]: This is a cool hook.

File: src/components/GMGrid.js (hunk 2 above)
Comment (spmcbride1201) [inline src/components/GMGrid.js]: I'm such an amazing hypocrite. lol

File: src/json/dashboards.json (hunk 3 above)
Comment (spmcbride1201) [inline src/json/dashboards.json]: It just ended up being crazy difficult to grok when it wasn't lined up.

File: src/store.js (hunk 4 above)
Comment (spmcbride1201) [inline src/store.js]: Pretty much the following: 
1. Cleaner syntax, especially for things other than get
2. HTTP Errors end up going to catch blocks, not then blocks
3. Ability to cancel requests

I was using fetch in this project previously, but I can't recall what the pain point was that pushed me to axios, my lib of choice.

File: src/components/GMGrid.js (hunk 2 above)
Comment (tilleryd) [inline src/components/GMGrid.js]: ha.  i'll keep ya honest

File: src/json/dashboards.json (hunk 3 above)
Comment (tilleryd) [inline src/json/dashboards.json]: no worries

File: src/store.js (hunk 4 above)
Comment (tilleryd) [inline src/store.js]: I'll give it a whirl, perhaps it'll make it into Corius2.

**Pull request merged**